### PR TITLE
[12.0] FIX mis_builder when user does not have time zone set.

### PR DESCRIPTION
**Pull request by eLBati**
Steps:

 - Install mis_builder
 - Empty time zone field of your user, if present
 - Create a MIS report template, following for example https://oca-mis-builder.readthedocs.io/en/latest/usage.html#reporting-on-non-accounting-data-queries

Get
`AttributeError: 'bool' object has no attribute 'upper'`

### CODE REVIEW

File: mis_builder/models/mis_report.py [hunk 1]
```diff
@@ -601,10 +601,9 @@ def _fetch_queries(self, date_from, date_to, get_additional_query_filter=None):
                     ]
                 )
             else:
-                datetime_from = _utc_midnight(date_from, self._context.get("tz", "UTC"))
-                datetime_to = _utc_midnight(
-                    date_to, self._context.get("tz", "UTC"), add_day=1
-                )
+                tz = self._context.get("tz", "UTC") or "UTC"
```
Comment (sbidoul) [inline mis_builder/models/mis_report.py]: Do you know where Odoo implies that no timezone means UTC ?

Comment (eLBati) [reply mis_builder/models/mis_report.py]: This https://github.com/odoo/odoo/blob/91d6598221ac50c839df749a2020846d89056daa/odoo/addons/base/models/ir_fields.py#L231 is an example

Comment (sbidoul) [reply mis_builder/models/mis_report.py]: Oh, thanks that's interesting. Maybe we could then use that `_input_tz` function here too, to be consistent with Odoo does ?

Comment (eLBati) [reply mis_builder/models/mis_report.py]: Right, done, thanks

**Comment by sbidoul** [comment]
I finished this with a newfragment in #450

**Pull request closed without merging**